sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -405,6 +409,246 @@ protected EventProcessor()
+
+                                    throw;
+                                }
+
+                                await Task.Delay(retryDelay.Value, cancellationSource.Token).ConfigureAwait(false);
+                            }
+                        }
+                    }
+                    catch (OperationCanceledException ex)
+                    {
+                        throw new TaskCanceledException(ex.Message, ex);
+                    }
+                    catch (DeveloperCodeException ex)
+                    {
+                        // Record that an exception was observed in developer-provided code, but consider it fatal and take no further
+                        // steps to handle or dispatch it.
+
+                        var message = string.Format(CultureInfo.InvariantCulture, Resources.DeveloperCodeExceptionMessageMask, ex.InnerException.Message);
+                        Logger.EventProcessorPartitionProcessingError(partition.PartitionId, Identifier, EventHubName, ConsumerGroup, message);
+
+                        ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
+                    }
+                    catch (Exception ex) when
+                        (ex is TaskCanceledException
+                        || ex is OutOfMemoryException
+                        || ex is StackOverflowException
+                        || ex is ThreadAbortException)
+                    {
+                        throw;
+                    }
+                    catch (Exception ex)
+                    {
+                        ++failedConsumerCount;
+                        capturedException = ex;
+                    }
+                    finally
+                    {
+                        try
+                        {
+                            if (consumer != null)
+                            {
+                                await consumer.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            Logger.EventProcessorPartitionProcessingError(partition.PartitionId, Identifier, EventHubName, ConsumerGroup, ex.Message);
+
+                            // Do not bubble the exception, as the consumer is being refreshed; failure to close this consumer is non-fatal.
+                        }
+                    }
+                }
+
+                // If there was an exception captured, then surface it.  Otherwise signal that cancellation took place.
+
+                if (capturedException != null)
+                {
+                    ExceptionDispatchInfo.Capture(capturedException).Throw();
+                }
+
+                throw new TaskCanceledException();
+            }
+
+            // Start processing in the background and return the processor
+            // metadata.
+
+            return new PartitionProcessor
```

Comment:
During the next round of implementation, this will be the structure that we track in the set of active partition processors, replacing the tuple used in the `EventProcessorClient` implementation, but serving the same purpose.